### PR TITLE
fix: insights-client failed, when --group was used

### PR DESCRIPTION
**Pull request by jirihnidek**
### All Pull Requests:

Check all that apply:

* [x] Have you followed the guidelines in our Contributing document, including the instructions about commit messages?
* [x] Is this PR to correct an issue?
* [ ] Is this PR an enhancement?

### Complete Description of Additions/Changes:

* Card ID: CCT-186
* When system was registered against Satellite server, and "insights-client --register" was triggered --group="tag" CLI option, then registration failed
* Satellite also contains some bug, and it always returns status code 200 and empty list regardless existence of given group.

### CODE REVIEW

**Comment by xiangce** [comment]
FYI, please don't worry about the failure of CI/CD, it was not caused by this PR and will be fixed once the new `uploader.json` is released. Please ignore it for now.

**Comment by jirihnidek** [comment]
I can see that Jenkins jobs failed, but I cannot see, why did it fail.

**Comment by xiangce** [comment]
> I can see that Jenkins jobs failed, but I cannot see, why did it fail.
Please ignore it too. It's a known issue mentioned in CCT-435. The QE team will fix it soon. Please ignore the other failures as well and start the review processes.

File: insights/client/connection.py [hunk 1]
```diff
@@ -573,35 +576,73 @@ def create_system(self, new_machine_id=False):
...
+            api_group_id = None
+            group_info = get_group.json()
+            if isinstance(group_info, dict):
+                try:
+                    results = group_info.get("results", None)
+                    if isinstance(results, list) and len(results) >= 1:
+                        group = results[0]
+                        if isinstance(group, dict):
+                            api_group_id = group.get("id", None)
+                    api_group_id = results
```
Comment (m-horky) [inline insights/client/connection.py:606]: You seem to be overwriting the value you have set one line above.

**Review by m-horky — CHANGES_REQUESTED** [review]
It's a shame we had to inflate the function by this much. It doesn't seem to be breaking anything by accident, and while it all looks ugly, there may be no way to save it.

Can you please revert the formatting changes that are committed with this change on other parts of the file?

**Review by m-horky — COMMENTED** [review]
Would it be possible to create tests for this? Mock out some example responses the server can return. To be honest, I'm kind of lost in this method now, I'd like to see the difference between hosted and Satellite.

**Comment by jirihnidek** [comment]
> Would it be possible to create tests for this? Mock out some example responses the server can return. To be honest, I'm kind of lost in this method now, I'd like to see the difference between hosted and Satellite.

I will try to do that.

**Review by m-horky — CHANGES_REQUESTED** [review]
Thank you. Would it be possible to add a test for the Satellite case as well? I'd like to have it covered as well. Just one more test that would show that response code 200 may not be enough.

**Comment by jirihnidek** [comment]
> Thank you. Would it be possible to add a test for the Satellite case as well? I'd like to have it covered as well. Just one more test that would show that response code 200 may not be enough.

What exactly do you mean to add test for the Satellite case?

**Comment by m-horky** [comment]
> Satellite also contains some bug, and it always returns status code 200 and empty list regardless existence of given group.

Something that would mock a response like that -- server returning 200 with empty list. The method in `connection.py` grew a lot, but the tests don't don't seem to be covering the edge case your patch was addressing.

**Review by m-horky — APPROVED** [review]
ACK

**Comment by xiangce** [comment]
May I know if this Pr ready for merge?

**Comment by m-horky** [comment]
@xiangce Hi, I believe we're waiting for QE to verify this patch. I'll ping you once it is ready.

**Comment by m-horky** [comment]
@xiangce Ready to be merged :+1:
